include/aspect/parameters.h

```diff
@@ -323,7 +324,14 @@ namespace aspect
     unsigned int                   n_expensive_stokes_solver_steps;
     double                         temperature_solver_tolerance;
     double                         composition_solver_tolerance;
-
+    unsigned int                   max_pre_newton_nonlinear_iterations;
+    unsigned int                   max_newton_line_search_iterations;
+
+    /// temporary place it in parameters.h till a better place is found
```

Comment:
I am not sure if this is the right place to put newton_residual and newton_theta, which are variables which are note given by the user, but used and updated only by the newton code